### PR TITLE
feat(tray): add start/stop recording options and global shortcut (WIP)

### PR DESCRIPTION
**Pull request by onyedikachi-david**
Fixes: #515 
/claim #515 

WIP

### CODE REVIEW

File: screenpipe-app-tauri/src-tauri/src/sidecar.rs [hunk 1]
```diff
@@ -369,11 +370,12 @@ fn spawn_sidecar(app: &tauri::AppHandle) -> Result<CommandChild, String> {
     Ok(child)
 }
 pub struct SidecarManager {
-    child: Option<CommandChild>,
+    pub child: Option<CommandChild>,
     last_restart: Instant,
     restart_interval: Arc<Mutex<Duration>>,
     restart_task: Option<JoinHandle<()>>,
     dev_mode: Arc<Mutex<bool>>,
+    pub is_recording: bool,
```
Comment (louis030195) [inline screenpipe-app-tauri/src-tauri/src/sidecar.rs:378]: this variable is not necessary, we just want to allow user to start or stop the sidecar with less friction than now (atm they need to go in screenpipe app and click on stop/start) we want to save them a few clicks/seconds on this

**Comment by louis030195** [comment]
ths seems to be #515 not #513

**Comment by onyedikachi-david** [comment]
> ths seems to be #515 not #513 

Yeah, right.

File: screenpipe-app-tauri/src-tauri/src/main.rs [hunk 2]
```diff
@@ -337,6 +384,86 @@ async fn main() {
...
             app.set_activation_policy(tauri::ActivationPolicy::Regular);
 
+
+            #[cfg(desktop)]
+            {
+                let toggle_recording_shortcut = Shortcut::new(Some(Modifiers::CONTROL | Modifiers::SHIFT), Code::KeyR);
+                let app_handle = app.handle().clone();
+                let event_app_handle = app_handle.clone(); // Clone for event emission
+
+                match app.handle().plugin(
```
Comment (louis030195) [inline screenpipe-app-tauri/src-tauri/src/main.rs:394]: can you make this less nested? really hard to read

**Comment by onyedikachi-david** [comment]
> can you make the shortcut changeable in settings like done in here? #524
> 
> if too hard i can make it before release

Yes, I thought of that too.

File: screenpipe-app-tauri/src-tauri/src/main.rs (hunk 2 above)
Comment (onyedikachi-david) [reply screenpipe-app-tauri/src-tauri/src/main.rs:394]: Okay

**Comment by onyedikachi-david** [comment]
Closing this due to conflicts.

**Pull request closed without merging**